app/controllers/partners/profiles_controller.rb

```diff
@@ -9,12 +9,18 @@ def update
         flash[:success] = "Details were successfully updated."
         redirect_to partners_profile_path
       else
+        handle_social_media_error_message if current_partner.errors.include?(:no_social_media_presence)
         render :edit
       end
     end
 
     private
 
+    def handle_social_media_error_message
+      current_partner.errors.delete(:no_social_media_presence)
```

Comment:
@cielf The reason that I have the "expect(page).not_to have_text("No social media presence must be accepted")" statement in the system test is to ensure that the original error message that says "No social media presence must be accepted" is getting deleted. Otherwise it will render on the page along with the correct error message. Are you sure that you want me to remove this expect statement from the test?